tools/tuf/test/create_repository.sh

```diff
@@ -64,10 +64,10 @@ for system in $SYSTEMS; do
     goarch_value="" # leave it empty to use the default for the system
     if [[ $system == "macos" ]]; then
         goose_value="darwin"
-	# for all platforms except Darwin, GOARCH is hardcoded to amd64 to
+	# for all platforms except Darwin, GOARCH defaults to amd64 to
 	# prevent cross compilation issues when building macOS arm64 binaries
 	# from Linux (CGO + libraries are required)
-	goarch_value="amd64"
+	goarch_value=${GOARCH:-amd64}
```

Comment:
so if we do this, and if you set `GOARCH`, it'll only affect building macOS, right? I would expect it to affect all platforms

Reply:
ah fair point, I think I was just focused on macos since that's why I created the PR originally 😅